lib/src/component_declaration/component_base.dart

```diff
@@ -377,7 +430,18 @@ abstract class UiProps
         return _validateChildren(children);
       });
 
-      return Function.apply(componentFactory, parameters);
+      final factory = componentFactory;
+      if (factory is ReactComponentFactoryProxy) {
+        // Use `build` instead of using emulated function behavior to work around DDC issue
+        // https://github.com/dart-lang/sdk/issues/29904
+        // Should have the benefit of better performance; TODO optimize type check?
+        return factory.build(props, invocation.positionalArguments);
+      } else {
+        var parameters = []
+          ..add(props)
+          ..addAll(invocation.positionalArguments);
+        return Function.apply(factory, parameters);
```

Comment:
Good point; though 99% of cases will use a `ReactComponentFactoryProxy`, it's in here for backwards compatibility.

I'll add some tests around this.